src/__tests__/UserInt.test.js

```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import '@testing-library/jest-dom/extend-expect';
+import userEvent from '@testing-library/user-event';
+import { act } from 'react-dom/test-utils';
+import App from '../App';
+
+describe('Testing user interactions on the main component (App.js).', () => {
+  describe('When user clicks on Home button', () => {
+    it('should render the home page', async () => {
+      act(() => {
+        render(<App />);
+      });
+      await userEvent.click(screen.getByRole('link', { name: /Home/i }));
+      expect(
+        screen.getByRole('heading', { name: /welcome to our page/i }),
+      ).toBeInTheDocument();
+    });
+  });
+});
```

Comment:
- [ ] @team Great work adding tests to some of your components, however, the project required testing of all components as shown below, kindly add tests for the remaining components.
![image](https://github.com/JOdiemo/Math-Magicians/assets/61408860/3979a18b-d604-427f-aa07-56c08a06eca6)